dev-lang/gnucobol/gnucobol-2.2.recipe

```diff
@@ -0,0 +1,123 @@
+SUMMARY="A free COBOL compiler"
+DESCRIPTION="GnuCOBOL (formerly OpenCOBOL) is a free COBOL compiler."
+HOMEPAGE="https://www.gnu.org/software/gnucobol/"
+COPYRIGHT="2001-2017 Free Software Foundation, Inc
+	Keisuke Nishida
+	Roger While
+	Ron Norman
+	Simon Sobisch
+	Edward Hart
+	Brian Tiffin
+	Sergey Kashyrin
+	Dave Pitts"
+LICENSE="GNU GPL v3
```

Comment:
Late note: that's actually wrong, it is GPLv3+ (same for LGPL3+)

Reply:
Looking at COPYING (says GNU GPL v3) and COPYING.LESSER (says GNU LGPLv3) is there any change noticeable to see this would be a 3+ (can't seem to find it)

Reply:
Thanks for asking and taking care.

Those two are the license files shipped with the source, the license that the software is under is written in the specific files. All below libcob [have][1]:

>    The GnuCOBOL runtime library is free software: you can redistribute it
>   and/or modify it under the terms of the GNU Lesser General Public License
>   as published by the Free Software Foundation, either version 3 of the
>   License, or (at your option) any later version.

and the files found [in cobc have][2]:

>    The GnuCOBOL compiler is free software: you can redistribute it
>   and/or modify it under the terms of the GNU General Public License
>   as published by the Free Software Foundation, either version 3 of the
>   License, or (at your option) any later version.

... and a good summary is found in the [README][3] (which may should include a note about the documentation license, which is just referred in the files that use it ):

> All programs except those in lib and libcob are distributed under
the GNU General Public License.  See COPYING for details.
> 
> Programs in lib and libcob are distributed under the GNU Lesser
General Public License.  See COPYING.LESSER for details.
> 
> For any copyright year range specified as YYYY-ZZZZ in this package,
that the range specifies every single year in that closed interval.

I don't know if the recipe format has a specific way to express license, many formats (like [npm / package.json][4]/nuspec use the [SPDX identifiers][5] (in this case), some use textual names (in this case often LGPLv3+) and again others like [PKGBUILD][6] use the name of an actual folder name that is shipped, for LGPv3+this would be LGPL3 (and the only has an extra one with "only" included), some use textual names (in this case often LGPLv3+), the [.spec Format][7].

... ok, found it, [haikuports recipes + License][8] says:

> Use the names from http://spdx.org/licenses/ if possible.

So that would be (for OpenCOBOL 2 and GnuCOBOL 2.2+):
~~~
GPL-3.0-or-later
LGPL-3.0-or-later
GFDL-1.3-or-later
~~

Thanks for the chance to update my license identifier knowledge ;-)

[1]:https://sourceforge.net/p/open-cobol/code/HEAD/tree/trunk/libcob/common.c
[2]:https://sourceforge.net/p/open-cobol/code/HEAD/tree/trunk/cobc/cobc.c
[3]:https://sourceforge.net/p/open-cobol/code/HEAD/tree/trunk/README
[4]:https://docs.npmjs.com/files/package.json#license
[5]:https://spdx.org/licenses/
[6]:https://wiki.archlinux.org/index.php/PKGBUILD#license
[7:]https://fedoraproject.org/wiki/Licensing:Main?rd=Licensing#Good_Licenses
[8]:https://github.com/haikuports/haikuports/wiki/HaikuPorter-BuildRecipes#license

Reply:
GPL v3+ is not a license, LICENSE content has to be a license.

Reply:
as noted above the license field, should be, according to the [haikuports recipes + License](https://github.com/haikuports/haikuports/wiki/HaikuPorter-BuildRecipes#license) contain the SPDX identifiers, one per line.

I guess I either did not understand your post (as a non-native speaker) or mis-read the docs?

Reply:
Here is the list: https://git.haiku-os.org/haiku/tree/data/system/data/licenses

Reply:
I see, so the old entry does not to seem wrong - but how should one understand the haikoports documentation linked above in this case? Full quote:

> The licenses should either be picked from the list in /system/data/licenses, or provided next to the recipe in a licenses folder as text file for inclusion inside the package.
Use the names from http://spdx.org/licenses/ if possible.

I understand it as "if possible do not use those in system/data/licenses but the spdx-identifier"; the big difference is that system/data/licenses has the license text while the spdx-identifier tell the actual license (which in this case is "or newer", something that can by definition be not part of the "not yet newer" license).

Reply:
The backstory here is that the system/data/licenses files are very old, and did not use spdx naming convention, and now we can't really change them because so many package rely on the current names.

So I think what is meant is "if you have to add your license in the package because it's not part of the standard set, use SPDX identifier as the filename". But indeed the wording isn't clear at all.

As for the "or later" clause, it is not technically part of the license itself. Maybe we can handle it like we do for GPL linking exceptions, as a separate file containing only the differences from the inital license. So in this case it would be something like "This software can, at your option, be used and redistributed under the terms of any later version of the GPL"?

Reply:
@pulkomandy

> The backstory here is that the system/data/licenses files are very old, and did not use spdx naming convention, and now we can't really change them because so many package rely on the current names.

Why not change the "old" ones to the SPDX identifier as the filename and then add a link from the "old" names pointing to the new ones? This should solve every conflict, shouldn't it?

Reply:
> The backstory here is that the system/data/licenses files are very old

What about updating the folder in general to include __common__ newer licenses, too?